src/features/domain-settings/hooks/useDomainSettingsForm.tsx

```diff
@@ -0,0 +1,227 @@
+import { ReactNode, useCallback, useState } from 'react';
+
+import {
+	steps,
+	FormStep,
+	FieldValues,
+	DetailsFormSubmit,
+	ConfirmActionType,
+} from '../DomainSettings.types';
+import { useDomainSettingsData } from '.';
+import { useWeb3, useZnsSdk } from '../../../lib/hooks';
+import {
+	CONFIRM_STEP_HEADER_TEXT,
+	LOADING_TEXT_CONTENT,
+} from '../DomainSettings.constants';
+import { useTransaction } from '@zero-tech/zapp-utils/hooks/useTransaction';
+
+import { ConfirmForm, DetailsForm } from '../Steps';
+import { Step, Wizard } from '@zero-tech/zui/components';
+
+export type UseDomainSettingsFormFormReturn = {
+	stepId: string;
+	formHeader: string;
+	formContent: ReactNode;
+	isTransactionLoading: boolean;
+	onStepBarUpdate: (step: Step) => void;
+};
+
+export const useDomainSettingsForm = (
+	zna: string,
+	onClose: () => void,
+): UseDomainSettingsFormFormReturn => {
+	const sdk = useZnsSdk();
+
+	const { account, provider } = useWeb3();
+	const { executeTransaction } = useTransaction();
+	const { domainId, metadata, isMetadataLocked } = useDomainSettingsData(zna);
+
+	const [stepId, setStepId] = useState(steps[0].id);
+	const [errorText, setErrorText] = useState<string>();
+	const [details, setDetails] = useState<DetailsFormSubmit>();
+	const [loadingStatusText, setLoadingStatusText] = useState<string>();
+	const [formHeader, setFormHeader] = useState<string>('My Domain Settings');
+	const [isTransactionLoading, setIsTransactionLoading] = useState<boolean>();
+	const [confirmActionType, setConfirmActionType] =
+		useState<ConfirmActionType>();
+
+	const onStepBarUpdate = (step: Step) => {
+		setStepId(step.id);
+	};
+
+	const onRestart = () => {
+		setStepId(steps[0].id);
+	};
+
+	// Set metadata form details
+	const onSubmitMetadata = ({
+		title,
+		description,
+		isMintable,
+		isBiddable,
+		gridViewByDefault,
+		customDomainHeader,
+		customDomainHeaderValue,
+	}: FieldValues): void => {
+		setDetails({
+			attributes: metadata?.attributes,
+			title,
+			description,
+			image: metadata?.image,
+			image_full: metadata?.image_full,
+			previewImage: metadata?.previewImage,
+			animation_url: metadata?.animation_url,
+			stakingRequests: metadata?.stakingRequests,
+			isBiddable,
+			isMintable,
+			gridViewByDefault,
+			customDomainHeader,
+			customDomainHeaderValue,
+		});
+	};
+
+	const onCheckMetadataLockStatus = useCallback(async () => {
+		const isDomainMetadataLocked = await sdk.isDomainMetadataLocked(
+			domainId,
+			provider.getSigner(),
+		);
+
+		return isDomainMetadataLocked;
+	}, [domainId, provider]);
```

Comment:
nit: would be good to use `react-query` here, but we can do that as an iteration